ConfigSpace/util.py

```diff
@@ -81,137 +81,157 @@ def get_one_exchange_neighbourhood(configuration: Configuration, seed: int) -> L
-                    number_of_sampled_neighbors += 1
-                except ValueError as e:
-                    pass
-
-                # Count iterations to not run into an infinite loop when
-                # sampling floats/ints and there is large amount of forbidden
-                #  values; also to find out if we tried to get a neighbor for
-                #  a categorical hyperparameter, and the only possible
-                # neighbor is forbidden together with another active
-                # value/default hyperparameter
-                iteration += 1
-
-    return neighbourhood
-
+                                else:
+                                    if not condition.evaluate(parents):
+                                        active = False
+                                        break
+
+                            if active and (current_value is None or
+                                           not np.isfinite(current_value)):
+                                default = current._inverse_transform(current.default)
+                                new_array[current_idx] = default
+                                children = configuration.configuration_space.get_children_of(
+                                    current.name)
+                                if len(children) > 0:
+                                    to_visit.extendleft(children)
+                                activated_values[current.name] = current.default
+
+                            if not active and (current_value is not None
+                                               or np.isfinite(current_value)):
+                                new_array[current_idx] = np.NaN
+
+                    try:
+                        # Populating a configuration from an array does not check
+                        #  if it is a legal configuration - check this (slow)
+                        new_configuration = Configuration(
+                            configuration.configuration_space, vector=new_array)
+                        new_configuration.is_valid_configuration()
+                        neighbourhood.append(new_configuration)
+                        number_of_sampled_neighbors += 1
+                    except ValueError as e:
+                        pass
+
+                    # Count iterations to not run into an infinite loop when
+                    # sampling floats/ints and there is large amount of forbidden
+                    #  values; also to find out if we tried to get a neighbor for
+                    #  a categorical hyperparameter, and the only possible
+                    # neighbor is forbidden together with another active
+                    # value/default hyperparameter
+                    iteration += 1
+            if len(neighbourhood) > 0:
+                neighbors_to_return[hp_name] = neighbourhood
+                np.random.shuffle(neighbors_to_return[hp_name])
```

Comment:
Same issue with the random state here. Also, if you shuffle here, you don't need to shuffle in the code above.